doc/unicode.md

```diff
@@ -37,35 +54,83 @@ NOT_CYRILLIC : [\P{Script=Cyrillic}] ;
 See [lexer-rules.md](lexer-rules.md#lexer-rule-elements) for more detail on Unicode
 escapes in lexer rules.
 
-# CharStreams and UTF-8
+## Migration
 
-## Java Target
-If your lexer grammar contains code points larger than `U+FFFF`, your
-lexer client code must open the file using `CharStreams.fromPath()` or
-equivalent in your runtime's language, or input values larger than
-`U+FFFF` will *not* match.
 
-For backwards compatibility, the existing `ANTLRInputStream` and
-`ANTLRFileStream` APIs only support Unicode code points up to `U+FFFF`.
+Code for **4.6** looked like this:
 
-The existing `TestRig` command-line interface supports all Unicode
-code points.
 
-## Other Targets
-Other language targets usually have their `ANTLRInputStream` extended to support the full Unicode range. See the target documentation for supported input encodings (e.g. UTF-8) and other related details.
+```java
+CharStream input = new ANTLRFileStream("myinputfile")
+JavaLexer lexer = new JavaLexer(input);
+CommonTokenStream tokens = new CommonTokenStream(lexer);
+```
 
-# Java Example
+Code for **4.7** looks like this:
 
-If you have generated a lexer named `UnicodeLexer`:
+```java
+CharStream input = CharStreams.fromFileName("inputfile");
+JavaLexer lexer = new JavaLexer(input);
+CommonTokenStream tokens = new CommonTokenStream(lexer);
+```
 
-```Java
-public static void main(String[] args) {
-  CharStream charStream = CharStreams.fromPath(Paths.get(args[0]));
-  Lexer lexer = new UnicodeLexer(charStream);
-  CommonTokenStream tokens = new CommonTokenStream(lexer);
-  tokens.fill();
-  for (Token token : tokens.getTokens()) {
-    System.out.println("Got token: " + token.toString());
-  }
-}
+Or, if you'd like to specify the file encoding:
+
+```java
+CharStream input = CharStreams.fromFileName("inputfile", StandardCharsets.UTF_16);
```

Comment:
Let's use something more realistic, like `Charset.forName("windows-1252")` (the default Windows text encoding for the US and Western Europe).